### PR TITLE
Perform a complete environment refresh for Studio install, build, deploy

### PR DESCRIPTION
**Pull request by joanise**
### PR Goal? 

Big jump in all the environment management for ReadAlongs/Studio:
 - Replace soon-to-be-deprecated `setup.py` and `requirements*.txt` by `pyproject.toml`
 - Move to the latest Heroku stack (heroku-24)
 - Move production on Heroku to the latest Python (3.13.1)

We still need a `requirements.txt` file for Heroku, but it's now effectively a lock file generated using `hatch`.

### Feedback sought? 

General review

### Priority? 

medium

### Tests added? 

yes, CI now has a dedicated reproducing the full Heroku way to build

### How to test? 

This version is currently deployed in preview mode on the heroku-24 stack at:

https://readalong-studio-heroku-24-488739ed89f8.herokuapp.com/api/v1/docs

(blocked as "Suspicious" on NRC networks, works elsewhere)

### Confidence? 

pretty high

### Version change? 

no

### CODE REVIEW

File: MANIFEST.in [hunk 1]
```diff
@@ -1,8 +0,0 @@
-recursive-include readalongs/templates *
-recursive-include readalongs/static *
```
Comment (roedoejet) [inline MANIFEST.in:2]: do we not need to include readalongs/static? I agree the templates and epub folders aren't here anymore.

Comment (roedoejet) [reply MANIFEST.in:2]: I don't see the static folder listed in the pyproject file either, but maybe it just get's included automatically or something? has this been tested?

File: readme-heroku.md [hunk 2]
```diff
@@ -1,16 +1,26 @@
 Our production Heroku deployment is controlled by the following files:
  - `Procfile`: tells Heroku what command to launch in each Dyno;
  - `runtime.txt`: tells Heroku which run-time engine to use (i.e., which version of Python);
+ - `requirements.txt`: tells Heroku what our production dependencies are;
+ - `bin/post_compile`: Heroku builds run this after doing `pip install -r requirements.txt`.
 
-   Heroku detects Python by default, but `runtime.txt` lets us specify/bump the version as needed;
- - `requirements.txt`: tells Heroku what our production dependencies are.
+Updating dependencies:
+ - Our dependencies are declared in `pyproject.toml`, edit this file to modify them.
```
Comment (roedoejet) [inline readme-heroku.md:8]:
```suggestion
 - Our dependencies are declared in `pyproject.toml`, edit `pyproject.toml` to modify them.
```

Comment (roedoejet) [reply readme-heroku.md:8]: "this file" is ambiguous

**Review by roedoejet — COMMENTED** [review]
I've got a question about the MANIFEST.in file and whether it's necessary anymore. Also, while looking at the docs in the demo API that is spun up I noticed that the help message for the `/langs` endpoint is incorrect. It gives this as an example:
```

[
    {"code": "alq", names: { "alq": "Anishinaabemowin", "_": "Algonquin" }},
    {"code": "atj", names: { "atj": "Nehiromowin", "_": "Atikamekw" }},
    {"code": "fra", names: { "fra": "Français", "_": "French" }},
    ...
]
```
but Français, Anishinaabemowin, and Nehiromowin don't actually show up, and I can't see any other examples of this. Also - by this logic, why is `_` not `eng`?

File: MANIFEST.in (hunk 1 above)
Comment (dhdaines) [reply MANIFEST.in:2]: Yes, Hatchling doesn't use `MANIFEST.in` from what I can tell: https://hatch.pypa.io/latest/why/#build-backend

So we need to double check to make sure the right things are being included in the source/wheel distributions.

Comment (joanise) [reply MANIFEST.in:2]: Good point, and thanks for the reminder @dhdaines! When Sam and I converted EveryVoice to `pyproject.toml`, we had to do some work to get the wheel and tar ball to contain the right files, and I didn't even test that yet with this PR. Better do that!

I've never understood what MANIFEST.in actually does, I have to admit... 🤷 I suppose I ought to!

**Comment by joanise** [comment]
> the help message for the `/langs` endpoint

I fixed the langs help message.

`_` was meant to be the key for the default display name, which is usually but not always English, and the other keys for names localized in specific languages, but in practice we have not done any of this localization work, so it's scaffolding for a future feature we have not implemented. I've adjusted the help to make that clear.

image

File: MANIFEST.in (hunk 1 above)
Comment (joanise) [reply MANIFEST.in:2]: Checks done: the wheel and tar ball get generated correctly.

**Review by roedoejet — APPROVED** [review]

**Pull request merged**